utils/signature/sigtest/factory.go

```diff
@@ -12,137 +12,74 @@ import (
 	"crypto/elliptic"
 	"crypto/rand"
 	"encoding/binary"
+	"io"
 	"math/big"
 	pseudorand "math/rand"
-	"strings"
 
-	"github.com/cockroachdb/errors"
 	"github.com/consensys/gnark-crypto/ecc/bn254"
 	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
 	"golang.org/x/crypto/sha3"
 
-	"github.com/hyperledger/fabric-x-committer/api/applicationpb"
 	"github.com/hyperledger/fabric-x-committer/utils"
 	"github.com/hyperledger/fabric-x-committer/utils/signature"
 )
 
-type (
-	// SignerVerifierFactory implements KeyFactory and instantiate verifiers and signers.
-	// This should only be used for evaluation and testing as it might use non-secure
-	// crypto methods.
-	SignerVerifierFactory struct {
-		KeyFactory
-		scheme signature.Scheme
-	}
-
-	// KeyFactory generates public and private keys.
-	KeyFactory interface {
-		NewKeys() (signature.PrivateKey, signature.PublicKey)
-		NewKeysWithSeed(int64) (signature.PrivateKey, signature.PublicKey)
-	}
-
-	// KeyFactory implementations.
-	dummyFactory struct{}
-	eddsaFactory struct{}
-	blsFactory   struct{}
-	ecdsaFactory struct{}
-)
-
-// String returns a string representation for logging.
-func (f *SignerVerifierFactory) String() string {
-	return f.Scheme()
-}
-
-// Scheme returns the factory scheme.
-func (f *SignerVerifierFactory) Scheme() signature.Scheme {
-	return f.scheme
-}
-
-// NewVerifier instantiate a verifier given a public key.
-func (f *SignerVerifierFactory) NewVerifier(key signature.PublicKey) (*signature.NsVerifier, error) {
-	v, err := signature.NewNsVerifier(&applicationpb.NamespacePolicy{
-		Rule: &applicationpb.NamespacePolicy_ThresholdRule{
-			ThresholdRule: &applicationpb.ThresholdRule{
-				Scheme:    f.scheme,
-				PublicKey: key,
-			},
-		},
-	}, nil)
-	return v, errors.Wrap(err, "error creating verifier")
-}
-
-// NewEndorser instantiate an endorser given a private key.
-func (f *SignerVerifierFactory) NewEndorser(key signature.PrivateKey) (*NsEndorser, error) {
-	return NewNsEndorser(f.scheme, key)
-}
-
-// NewSignatureFactory instantiate a SignerVerifierFactory.
-func NewSignatureFactory(scheme signature.Scheme) *SignerVerifierFactory {
-	f, err := getFactory(scheme)
-	if err != nil {
-		panic(err.Error())
+// NewKeys generate private and public keys.
+// This should only be used for evaluation and testing as it might use non-secure crypto methods.
+func NewKeys(scheme signature.Scheme) (signature.PrivateKey, signature.PublicKey) {
```

Comment:
Great. The removal of factory pattern simplifies the code as well as improves the readability, IMO. 

We can name this `NewKeyPair` instead of `NewKeys`. I understand this is a tech debt from the existing code. The same suggestion is applicable for all other `xxxNewKeys()`.